### PR TITLE
PSDEVOPS-3851 brew tag relations

### PR DESCRIPTION
**Pull request by jasinner**
This adds BREW_TAG type to relations. It also adds several fields to ProductStream model for all the existing methods of ingesting builds and relating them to products in Deptopia/product-definitions.

We need to adjust the existing compose collector to use the new composes field of ProductStream to relate those.

### CODE REVIEW

File: corgi/core/models.py [hunk 1]
```diff
@@ -443,6 +443,13 @@ class ProductStream(ProductModel, TimeStampedModel):
 
     cpe = models.CharField(max_length=1000, default="")
 
+    # NOTE brew_tags and yum_repositories values shouldn't be exposed outside of Red Hat
+    brew_tags = models.JSONField(default=dict)
```
Comment (JimFuller-RedHat) [inline corgi/core/models.py:447]: It is fine and good to add specific attributes to ProductStream ... mentally I read the ProductModel base
as RedHatProductModel.

File: corgi/core/models.py [hunk 2]
```diff
@@ -561,6 +568,8 @@ class ProductComponentRelation(TimeStampedModel):
     class Type(models.TextChoices):
         ERRATA = "ERRATA"
         COMPOSE = "COMPOSE"
+        BREW_TAG = "BREW_TAG"
```
Comment (JimFuller-RedHat) [inline corgi/core/models.py:571]: nice ...

File: corgi/tasks/brew.py [hunk 3]
```diff
@@ -300,3 +307,24 @@ def save_module(softwarebuild, build_data) -> ComponentNode:
...
+
+
+def load_brew_tags() -> None:
+    for ps in ProductStream.objects.all():
+        brew = Brew()
+        for brew_tag, inherit in ps.brew_tags.items():
+            try:
+                builds = brew.get_builds_with_tag(brew_tag, inherit)
+            except GenericError as exc:
+                logger.error("Got error getting builds for %s: %s", brew_tag, exc)
```
Comment (JimFuller-RedHat) [inline corgi/tasks/brew.py:319]: Is it intentional to keep running a task and just log the error ? Alternately celery would trap this right ?

**Comment by JimFuller-RedHat** [comment]
apart from minor comments, this is a nice direction and very handy LGTM

File: corgi/tasks/brew.py (hunk 3 above)
Comment (jasinner) [reply corgi/tasks/brew.py]: Ok, removed the try/catch from here, and letting Celery retry this in case of error.

**Pull request merged**